tests/fixtures/__init__.py

```diff
@@ -72,18 +89,45 @@ def non_mocked_hosts() -> list[str]:
     return ["test"]
 
 
-@async_fixture(name="client")
-async def fixture_client(mongodb_fixture) -> AsyncGenerator[AsyncClient, None]:
-    """Get test client for the work package service"""
+async def publish_datasets(container: Container) -> None:
+    """Publish datasets to populate the database."""
+    config = container.config()
+    async with KafkaEventPublisher.construct(config=config) as event_publisher:
+        await event_publisher.publish(
+            payload=DATASET_OVERVIEW_EVENT.dict(),
+            type_="metadata_dataset_overview",
+            key="test_key",
+            topic="metadata",
+        )
+
+    event_subscriber = await container.event_subscriber()
+    await asyncio.wait_for(event_subscriber.run(forever=False), timeout=5)
+
+
+@async_fixture(name="container")
+async def fixture_container(
+    mongodb_fixture, kafka_fixture
+) -> AsyncGenerator[Container, None]:
+    """Populate database and get configured container"""
 
     config = Config(
         auth_key=AUTH_KEY_PAIR.export_public(),
         download_access_url="http://access",
         work_package_signing_key=SecretStr(SIGNING_KEY_PAIR.export_private()),
+        **kafka_fixture.config.dict(),
         **mongodb_fixture.config.dict(),
     )
 
-    async with get_container(config=config):
-        api = get_rest_api(config=config)
-        async with AsyncClient(app=api, base_url="http://test") as client:
-            yield client
+    async with get_container(config=config) as container:
+        await publish_datasets(container)
+        yield container
+
+
+@async_fixture(name="client")
+async def fixture_client(container) -> AsyncGenerator[AsyncClient, None]:
```

Comment:
```suggestion
async def fixture_client(container: Container) -> AsyncGenerator[AsyncClient, None]:
```

Reply:
It might be worth to harmonize the style of writing fixtures with the file services. Added this to the refactoring list.